cmds/core/sluinit/uinit.go

```diff
@@ -110,3 +121,46 @@ func unmountAndExit() {
 	time.Sleep(5 * time.Second) // let queued up debug statements get printed
 	os.Exit(1)
 }
+
+// scanIscsiDrives calls dhcleint to parse cmdline and
+// iscsinl to mount iscsi drives.
+func scanIscsiDrives() error {
+
+	val, ok := cmdline.Flag("netroot")
+	if !ok {
+		return errors.New("netroot flag is not set")
+	}
+	slaunch.Debug("netroot flag is set with val=%s", val)
+
+	var n *net.TCPAddr
+	n, volume, err := dhclient.ParseISCSIURI(val)
+	if err != nil {
+		return fmt.Errorf("dhclient ISCSI parser failed err=%v", err)
+	}
+
+	ip := n.IP.String() + ":" + "3260"
```

Comment:
Why are you manually setting the port to 3260? This would override a user-specified value

Reply:
good catch. I went through this design decision last night.. Notice that calling IP.String and Itoa in sluinit would be simply reversing whats done in dhclient parseISCSIURI, namely ip = net.ParseIP(tok) and pv, err := strconv.Atoi(tok)....
Ideally, we would want to not have this :: do A, do B, undo A, undo B behavior. ..But that would mean breaking down parseISCSIURI into two parts...one part returns tokens as strings and next part wraps these strings into net.TCPAddr...You cool with that ?

Reply:
I have updated the PR with the approach i suggested in my comment above.. Please let me know if you prefer method 2 or method 1...

Reply:
I'm not sure I see a problem with going from string to struct back to string. It's just a canonical representation, it should work out? And iscsinl probably parses it back into a TCPAddr internally, so ideally we'd just make that a TCPAddr instead...

Reply:
okay, I went back to method1, and used itoa to convert port number received from parseISCSIURI to string so that it can be passed to iscsinl. All set.